tests/toranj/build.sh

```diff
@@ -221,6 +224,7 @@ case ${build_config} in
             -DOT_THREAD_VERSION=1.4 -DOT_APP_CLI=ON -DOT_APP_NCP=OFF -DOT_APP_RCP=OFF \
             -DOT_15_4=ON -DOT_TREL=ON \
             -DOT_PLATFORM_KEY_REF=${ot_plat_key_ref} \
+            -DOT_SIMULATION_INFRA_IF=ON \
```

Comment:
Instead of here, how about adding in the config headers? e.g. in `./tests/toranj/openthread-core-toranj-config-simulation.h`.

We include configs that differ based on the `{build_config}` within the `build.sh` script. All other configs are defined in the related `toranj` header files.